app/javascript/entrypoints/two_factor_authentication.ts

```diff
@@ -0,0 +1,199 @@
+import * as WebAuthnJSON from '@github/webauthn-json';
+import axios, { AxiosError } from 'axios';
+
+import ready from '../mastodon/ready';
+
+import 'regenerator-runtime/runtime';
+
+type PublicKeyCredentialCreationOptionsJSON = Parameters<
+  (typeof WebAuthnJSON)['create']
+>[0]['publicKey'];
+
+function exceptionHasAxiosError(
+  error: unknown,
+): error is AxiosError<{ error: unknown }> {
+  return (
+    error instanceof AxiosError &&
+    typeof error.response?.data === 'object' &&
+    'error' in error.response.data
+  );
+}
+
+function logAxiosResponseError(error: unknown) {
+  if (exceptionHasAxiosError(error)) console.error(error);
+}
```

Comment:
Ditto.

Reply:
I am not sure what we could change here? `exceptionHasAxiosError` is a [type predicate](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates) to narrow down the type of an exception

Reply:
Yeah, this is a bit boilerplate-y but I don't really see how it could be made more readable.